### ISSUE TITLE
Fatal LaTeX errors can cause cause assignment errors in Scriptorium.

### ISSUE THREAD

**Issue by newtonj** [jasedit/scriptorium#12]
A fatal LaTeX error can break error output statements inside Scriptorium. The stack trace and causative error are included below.

```
Traceback (most recent call last):
  File "/usr/local/bin/scriptorium", line 11, in <module>
    load_entry_point('scriptorium==2.0.0', 'console_scripts', 'scriptorium')()
  File "/usr/local/lib/python2.7/dist-packages/scriptorium-2.0.0-py2.7.egg/scriptorium/__main__.py", line 78, in main
    args.func(args)
  File "/usr/local/lib/python2.7/dist-packages/scriptorium-2.0.0-py2.7.egg/scriptorium/__main__.py", line 12, in make
    pdf = scriptorium.to_pdf(args.paper, use_shell_escape=args.shell_escape)
  File "/usr/local/lib/python2.7/dist-packages/scriptorium-2.0.0-py2.7.egg/scriptorium/papers.py", line 76, in to_pdf
    print('LaTeX conversion failed with the following output:\n', output)
UnboundLocalError: local variable 'output' referenced before assignment
```

```
LaTeX Warning: Citation 'mrx05' on page 1 undefined on input line 16.

pdfTeX warning (ext4): destination with the same identifier (name{page.1}) has
been already used, duplicate ignored
<to be read again>
                   \relax
l.18 \chapter
             {Bibliography} [1

]
Chapter 2.

(/home/hakujin/scriptorium/bin/../templates/./simple_templates/report/footer.te
x

! Package biblatex Error: '\bibliographystyle' invalid.

See the biblatex package documentation for explanation.
Type  H <return>  for immediate help.
 ...

l.4   \bibliographystyle{ieeetr}

Here is how much of TeX's memory you used:
 10908 strings out of 495000
 178786 string characters out of 6181170
 614717 words of memory out of 5000000
 14062 multiletter control sequences out of 15000+600000
 8501 words of font info for 30 fonts, out of 8000000 for 9000
 14 hyphenation exceptions out of 8191
 37i,6n,32p,992b,729s stack positions out of 5000i,500n,10000p,200000b,80000s

!  ==> Fatal error occurred, no output PDF file produced!
```

**Comment by newtonj** [comment]
Actual error fixed by [332f721](https://github.com/jasedit/scriptorium/commit/332f721445ad4e7c639a94d4c2b363bdf8cb29ae), but LaTeX errors are output as an unreadable (even more than usual) mess.

**Comment by newtonj** [comment]
Eldrich runes are copied below. May or may not be some esoteric python tuple.

```
('LaTeX conversion failed with the following output:\n', "This is pdfTeX, Version 3.14159265-2.6-1.40.16 (TeX Live 2015/Debian) (preloaded format=pdflatex)\n restricted \\write18 enabled.\nentering extended mode\n(./paper.tex\nLaTeX2e <2015/01/01>\nBabel <3.9l> and hyphenation patterns for 2 languages loaded.\n\n(/home/hakujin/scriptorium/bin/../templates/./simple_templates/report/setup.tex\n(/usr/share/texlive/texmf-dist/tex/latex/base/report.cls\nDocument Class: report 2014/09/29 v1.4h Standard LaTeX document class\n(/usr/share/texlive/texmf-dist/tex/latex/base/size10.clo))\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphics.sty\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/trig.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/graphics.cfg)\n(/usr/share/texlive/texmf-dist/tex/latex/pdftex-def/pdftex.def\n(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/infwarerr.sty)\n(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/ltxcmds.sty)))\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphicx.sty\n(/usr/share/texlive/texmf-dist/tex/latex/graphics/keyval.sty))\n(/usr/share/texlive/texmf-dist/tex/latex/url/url.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/hyperref/hyperref.sty\n(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/hobsub-hyperref.sty\n(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/hobsub-generic.sty))\n(/usr/share/texlive/texmf-dist/tex/generic/ifxetex/ifxetex.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/auxhook.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/kvoptions.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/hyperref/pd1enc.def)\n(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/hyperref.cfg))\n\nPackage hyperref Message: Driver (autodetected): hpdftex.\n\n(/usr/share/texlive/texmf-dist/tex/latex/hyperref/hpdftex.def\n(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/rerunfilecheck.sty))\n(/usr/share/texlive/texmf-dist/tex/latex/biblatex/biblatex.sty\n(/usr/share/texlive/texmf-dist/tex/latex/biblatex/biblatex2.sty\n(/usr/share/texlive/texmf-dist/tex/latex/etoolbox/etoolbox.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/logreq/logreq.sty\n(/usr/share/texlive/texmf-dist/tex/latex/logreq/logreq.def))\n(/usr/share/texlive/texmf-dist/tex/latex/base/ifthen.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/biblatex/blx-dm.def)\n(/usr/share/texlive/texmf-dist/tex/latex/biblatex/blx-compat.def)\n(/usr/share/texlive/texmf-dist/tex/latex/biblatex/biblatex.def)\n(/usr/share/texlive/texmf-dist/tex/latex/biblatex/blx-natbib.def)\n(/usr/share/texlive/texmf-dist/tex/latex/biblatex/bbx/numeric.bbx\n(/usr/share/texlive/texmf-dist/tex/latex/biblatex/bbx/standard.bbx))\n(/usr/share/texlive/texmf-dist/tex/latex/biblatex/cbx/numeric.cbx)\n(/usr/share/texlive/texmf-dist/tex/latex/biblatex/biblatex.cfg)))\n(/usr/share/texlive/texmf-dist/tex/latex/geometry/geometry.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/biblatex/lbx/english.lbx) (./paper.aux\n) (/usr/share/texlive/texmf-dist/tex/context/base/supp-pdf.mkii\n[Loading MPS to PDF converter (version 2006.09.02).]\n) (/usr/share/texlive/texmf-dist/tex/latex/oberdiek/epstopdf-base.sty\n(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/grfext.sty)\n(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg))\n(/usr/share/texlive/texmf-dist/tex/latex/hyperref/nameref.sty\n(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/gettitlestring.sty))\n(./paper.out) (./paper.out)\nNo file paper.bbl.\n*geometry* driver: auto-detecting\n*geometry* detected driver: pdftex\n[1{/var/lib/texmf/fonts/map/pdftex/updmap/pdftex.map}])\nChapter 1.\n\nOverfull \\hbox (26.63655pt too wide) in paragraph at lines 9--9\n[]\\OT1/cmr/bx/n/14.4 Optical MEMS Pres-sure Sen-sors In-cor-po-rat-ing Dual Wav\neg-\n\nLaTeX Warning: Citation 'mrx05' on page 1 undefined on input line 16.\n\npdfTeX warning (ext4): destination with the same identifier (name{page.1}) has \nbeen already used, duplicate ignored\n<to be read again> \n                   \\relax \nl.18 \\chapter\n             {Bibliography} [1]\nChapter 2.\n\n(/home/hakujin/scriptorium/bin/../templates/./simple_templates/report/footer.te\nx\n\n! Package biblatex Error: '\\bibliographystyle' invalid.\n\nSee the biblatex package documentation for explanation.\nType  H <return>  for immediate help.\n ...                                              \n                                                  \nl.4   \\bibliographystyle{ieeetr}\n                                \n!  ==> Fatal error occurred, no output PDF file produced!\nTranscript written on paper.log.\n")
```

**Comment by newtonj** [comment]
The directory used for repro is included below.
[advanced_instrumentation.tar.gz](https://github.com/jasedit/scriptorium/files/475479/advanced_instrumentation.tar.gz)

**Comment by newtonj** [comment]
Horrifying gibberish is fixed by [cc6dd11](https://github.com/jasedit/scriptorium/commit/cc6dd11c824e1ab814eb704db004ed86d9bea8c8). Issue is now fully resolved.
